kafka-source/src/main/java/org/springframework/cloud/stream/module/kafka/source/KafkaConfigurationProperties.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.kafka.source;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Properties related to connecting to Kafka broker.
+ *
+ * @author Soby Chacko
+ */
+@ConfigurationProperties
+public class KafkaConfigurationProperties {
+
+	private String zkConnect = "localhost:2181";
+
+	private String zkConnectionTimeout = "6000";
+
+	private String zkSessionTimeout = "6000";
+
+	private String zkSyncTime = "2000";
+
+	private String keyDecoder;
+
+	private String payloadDecoder;
+
+	private Map<String, String> partitions = new HashMap<>();
+
+	private Map<String, Map<Integer, Long>> initialOffsets = new HashMap<>();
+
+	private String[] topics;
+
+	private int concurrency;
+	private int maxFetch;
+	private int stopTimeout;
+	private int queueSize;
+
+	public int getQueueSize() {
+		return queueSize;
+	}
+
+	public void setQueueSize(int queueSize) {
+		this.queueSize = queueSize;
+	}
+
+	public int getStopTimeout() {
+		return stopTimeout;
+	}
+
+	public void setStopTimeout(int stopTimeout) {
+		this.stopTimeout = stopTimeout;
+	}
+
+	public int getMaxFetch() {
+		return maxFetch;
+	}
+
+	public void setMaxFetch(int maxFetch) {
+		this.maxFetch = maxFetch;
+	}
+
+	public int getConcurrency() {
+		return concurrency;
+	}
+
+	public void setConcurrency(int concurrency) {
+		this.concurrency = concurrency;
+	}
+
+	public String[] getTopics() {
+		return topics;
+	}
+
+	public void setTopics(String[] topics) {
+		this.topics = topics;
+	}
+
+	public Map<String, String> getPartitions() {
+		return partitions;
+	}
+
+	public void setPartitions(Map<String, String> partitions) {
+		this.partitions = partitions;
+	}
+
+	public String getZkConnect() {
```

Comment:
Well, I mean something like this:
```
@IntegrationTest({"topics = foo", "zkConnect="})
```
when we just can't fall back, because of end-user interaction.